File: gymanager/lib/src/models/member_model.dart
class Member {
  final String id;
  final String name;
  final String gender;
  final DateTime dob;
  final DateTime membershipEndDate;
  final String? photoUrl;
  final String? trainerName; // New field for the trainer's name

  Member({
    required this.id,
    required this.name,
    required this.gender,
    required this.dob,
    required this.membershipEndDate,
    this.photoUrl,
    this.trainerName,
  });

  factory Member.fromJson(Map<String, dynamic> json) {
    // Handle the populated trainer data
    String? tName;
    if (json['trainerId'] != null && json['trainerId'] is Map) {
      tName = json['trainerId']['name'];
    }

    return Member(
      id: json['_id'],
      name: json['name'],
      gender: json['gender'],
      dob: DateTime.parse(json['dob']),
      membershipEndDate: DateTime.parse(json['membershipEndDate']),
      photoUrl: json['photoUrl'],
      trainerName: tName,
    );
  }
}

File: backend_server/models/member.model.js
const mongoose = require('mongoose');

const memberSchema = mongoose.Schema(
  {
    userId: { // The gym owner who created this member
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: 'User',
    },
    name: { type: String, required: true },
    gender: { type: String, required: true },
    dob: { type: Date, required: true }, // Date of Birth
    height: { type: Number },
    weight: { type: Number },
    photoUrl: { type: String },
    trainerId: { // Optional assigned trainer
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Trainer',
      default: null,
    },
    membershipStartDate: { type: Date, required: true },
    membershipEndDate: { type: Date, required: true },
  },
  {
    timestamps: true,
  }
);

const Member = mongoose.model('Member', memberSchema);
module.exports = Member;